codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/http/HttpBindingGenerator.kt

```diff
@@ -386,17 +411,37 @@ class HttpBindingGenerator(
                     })
                     """,
                 )
-            else -> rustTemplate(
-                """
-                if $parsedValue.len() > 1 {
-                    Err(#{header_util}::ParseError::new_with_message(format!("expected one item but found {}", $parsedValue.len())))
+            else -> {
+                if (targetShape is ListShape) {
+                    // This is a constrained list shape and we must therefore be generating a server SDK.
```

Comment:
Should we match on the `RustType.MaybeContrained` type to be absolutely sure, and have a catch all `else` that panics since the codegen encountered something it doesn't understand?

Reply:
This is not a `RustType.MaybeConstrained`, it's a `RustType.Opaque` (created by `UnconstrainedCollectionGenerator`). `MaybeConstrained` is only used in server builder members.